build-tools/packages/build-infrastructure/src/fluidRepo.ts

```diff
@@ -235,3 +237,30 @@ export function getAllDependenciesInRepo(
 		workspaces: [...workspaces],
 	};
 }
+
+export async function setDependencyRange(
+	packagesToUpdate: IPackage[],
+	dependencies: IPackage[],
+	dependencyRange: InterdependencyRange
+): Promise<void> {
+	for (const pkg of packagesToUpdate) {
+		for (const dep of pkg.combinedDependencies) {
+			if (dependencies.some(d => d.name === dep.name)) {
+				const depRange = typeof dependencyRange === "string"
+					? dependencyRange
+					: (dependencyRange instanceof semver.SemVer ? dependencyRange.version : undefined);
+
+				const depName = dep.name;
+				// Update the version in packageJson
+				if ((pkg.packageJson.dependencies?.[depName]) !== undefined) {
+					pkg.packageJson.dependencies[depName] = depRange;
+				} else if ((pkg.packageJson.devDependencies?.[depName]) !== undefined) {
+					pkg.packageJson.devDependencies[depName] = depRange;
+				} else if ((pkg.packageJson.peerDependencies?.[depName]) !== undefined) {
+					pkg.packageJson.peerDependencies[depName] = depRange;
+				}
+			}
+		}
+		await pkg.savePackageJson();
```

Comment:
Consider using a pattern like this:

```ts
// collect the "save" promises in an array
const savePromises: Promise<void>[] = [];

// in your loop
savePromises.push(pkg.savePackageJson());

// after the loop
await Promise.all(savePromises);
```

That will parallelize the saves, which is safe since they're all independent.